sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -246,7 +247,27 @@ case class LoadDataCommand(
     val loadPath =
       if (isLocal) {
         val uri = Utils.resolveURI(path)
-        if (!new File(uri.getPath()).exists()) {
+        val filePath = uri.getPath()
+        val exists = if (filePath.contains("*")) {
+          val splitPath = filePath.split(File.separator)
+          val filePattern = splitPath.last
+          val dir = splitPath.dropRight(1).mkString(File.separator)
+          if (dir.contains("*")) {
+            throw new AnalysisException(
+              s"LOAD DATA input path allows only filename wildcard: $path")
+          }
+
+          val files = new File(dir).listFiles()
+          if (files == null) {
+            false
+          } else {
+            val matcher = FileSystems.getDefault.getPathMatcher("glob:" + filePattern)
```

Comment:
Nit, you might get one reference to `FileSystems.getDefault` and reuse it rather than keep retrieving it.